src/pages/Booking/MusicalBooking/TheatreItem.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { useBookingStore } from '../store/store';
+import { shallow } from 'zustand/shallow';
+
+function TheatreItem({ handleDynamicFetch, id, image, name, description }) {
+  const [selectedTheatre, setSelectedTheatre, setSelectedTime] =
+    useBookingStore(
+      state => [
+        state.bookingState.selectedTheatre,
+        state.setTheatre,
+        state.setTime,
+      ],
+      shallow
+    );
+
+  const updateTheatre = theatre => {
+    const { id, name } = theatre;
+    setSelectedTheatre({
+      id: id,
+      name: name,
+    });
+    setSelectedTime(null);
+  };
+
+  return (
+    <TheatreItemContainer
+      isSelected={id === selectedTheatre?.id ? true : false}
+      onClick={() => {
+        handleDynamicFetch('theatreId', id);
+        updateTheatre({
+          id,
+          name,
+        });
+      }}
+    >
```

Comment:
```suggestion
const handleTheatre = ( )=>{
        handleDynamicFetch('theatreId', id);
        updateTheatre({id,name})
        }
...
      onClick={handleTheatre}
    >
```

id,name은 props로 받고있고,
이 함수에서 매개변수를 전달해주는건 없는것같아서
이벤트함수도 분리해서 함수 자체만 껴주는건 어떨까요.